### ISSUE TITLE
Consider deprecating the opencensus exporter and receiver

### ISSUE THREAD

**Issue by atoulme** [open-telemetry/opentelemetry-collector-contrib#36791]
### Component(s)

exporter/opencensus, receiver/opencensus

### Describe the issue you're reporting

The OpenCensus functionality has been around for a bit but is now superceded by OTLP.

I propose that we start deprecating those components, and set a deadline for removal.

The calendar would work as follows:
* Mark the components as deprecated
* 6 months later, delete the components.
This will give time to anyone using the components to make plans. We can make the duration of the deprecation longer based on community input.

Please provide your feedback as comments and reactions. Before we proceed, I would like to make sure we receive a healthy number of 👍 from approvers and maintainers of the repository.

**Comment by atoulme** [comment]
I have not heard back on this plan yet. I will move to open a first PR to deprecate the components and publish a deprecation and removal schedule for review by approvers.

**Comment by dashpole** [comment]
I support deprecating both.  6 months seem like a reasonable time period to migrate off to me.

**Comment by dprotaso** [comment]
Hi - any chance we can increase the buffer a little bit? Say till the end of 2025?

**Comment by atoulme** [comment]
@dprotaso for both exporter and receiver? Is there a reason to keep those components around, can you try OTLP?

**Comment by dprotaso** [comment]
hey @atoulme all good - Knative migrated away from OpenCensus in the last release.

I was suggesting an extension so our end users (on older releases) could have some more grace period but I think it's fine.

**Comment by alexchowle** [comment]
Unfortunately, this deprecation is going to hurt my stack. My managed service mesh version in my cloud provider (GCP Anthos Service Mesh) only supports OpenCensus.

**Comment by dprotaso** [comment]
@alexchowle I'd talk to a google rep about that

**Comment by alexchowle** [comment]
> [@alexchowle](https://github.com/alexchowle) I'd talk to a google rep about that

I have. Now I'm here. I'm not saying it couldn't support something like OTLP, but that's a large change.

**Comment by atoulme** [comment]
@alexchowle thanks for the heads up, I've pinged a few folks who work in that area. Hopefully they let us know the plan.

**Comment by alexchowle** [comment]
> [@alexchowle](https://github.com/alexchowle) thanks for the heads up, I've pinged a few folks who work in that area. Hopefully they let us know the plan.

I don't look after the service mesh but, as I understand it, the managed version of Anthos Service Mesh we're on only supports OpenCensus. 

FWIW - Google's replacement product which is called "Cloud Service Mesh" doesn't support any egress of trace spans; it's only compatible with Google Cloud Trace.

**Comment by joshdaws0n** [comment]
I'd find it massively valuable if keeping the OpenCensus receiver was possible.

**Comment by alexchowle** [comment]
@atoulme - has the component been removed from the release, regardless?

https://github.com/open-telemetry/opentelemetry-collector-releases/releases/tag/v0.132.0

> Breaking changes 🛑
releases: Fix artifact checksum files being incomplete and split between Linux/Darwin and Windows. (https://github.com/open-telemetry/opentelemetry-collector-releases/issues/1002, https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1086)
This fixes a bug that occurred because releases for Windows and other Platforms are split.
The Windows release always overwrote the checksum files already created for Linux and Darwin and so, only
the Windows checksums were accessible in the end.
This change is breaking because checksum files for Linux/Darwin and Windows are now split.
otelcol: Remove deprecated opencensus exporter and receiver (https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1056)
otelcol-contrib: Remove deprecated opencensus exporter and receiver (https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1056)
otelcol-k8s: Remove deprecated opencensus exporter and receiver (https://github.com/open-telemetry/opentelemetry-collector-releases/pull/1056)

**Comment by atoulme** [comment]
It's removed from contrib, yes.

**Comment by alexchowle** [comment]
> It's removed from contrib, yes.

OK. Am I right in thinking that my only remaining option is to compile my own Collector against a frozen-in-time OpenCensus Receiver?

**Comment by alexchowle** [comment]
@atoulme  - It's worth me explaining my situation in a little more detail...

Anthos Service Mesh is Google's managed Istio Service Mesh. It does not offer all of Istio's capability, however. An example being the ability to export Traces in OTLP format. It only provides OpenCensus as an export format. I need to receive the Mesh's Spans because of two reasons:

1. The Mesh participates in existing Distributed Traces. It will appreciate the W3C Trace header and create Spans for its ingress and egress of Services on the Mesh. If I cannot obtain these additional Mesh Spans, the Trace lineage is broken which leads to fragmented Traces in my backend Trace platform.
2. I have a mixture of applications, some of which have OTel instrumentation, and some without. For those without, the Mesh's Spans are paramount because of the participation in existing Traces (point # 1 above) -- I get better observability in my distributed system by obtaining the Mesh's Spans.

To date, the OTel Collector "k8s" and "contrib" distributions have proven to be a fantastic "Swiss army knife" for me because it can ingest and produce such a variety of formats. This is why it's important to me to have a build with said OpenCensus Receiver; without it I have nothing but fragmented Traces.

If my only workaround is to build my own Collector against an ageing, static OpenCensus Receiver (and cope with the gradual rewrites of its dependencies via **me** changing its `go.mod` ) then that's something I need to consider. I need to buy time for my Service Mesh to be replaced to something more OTel-friendly. (FWIW - its Google replacement is called "Cloud Service Mesh" which has **no** support for exporting Mesh Spans :( )

Is the removal of this component more about "it's time to sunset OpenCensus in OpenTelemetry because OTel replaces OpenCensus", or is there a definite cost in keeping it around and maintaining it in this repo and the Collector builds?

**Comment by atoulme** [comment]
> > It's removed from contrib, yes.
> 
> OK. Am I right in thinking that my only remaining option is to compile my own Collector against a frozen-in-time OpenCensus Receiver?

No, the receiver is still in the contrib tree and has been released as part of 0.132.0. You can build your own collector with that version.

**Comment by atoulme** [comment]
> Is the removal of this component more about "it's time to sunset OpenCensus in OpenTelemetry because OTel replaces OpenCensus", or is there a definite cost in keeping it around and maintaining it in this repo and the Collector builds?

We discussed this at the SIG meeting this morning. Please look at the notes/watch the recording for context.

The OpenCensus project predates OpenTelemetry and the maintainers made the commitment to maintain it when the collector project was established. After years of maintenance, there is no longer interest from maintainers and approvers to maintain this component, and we are deprecating it with the following possible outcomes:

* It gets removed from our source tree
* Someone steps up to be a codeowner of the component

We have followed process ; after 6 months in deprecation state, this component is being removed.

**Comment by alexchowle** [comment]
> > Is the removal of this component more about "it's time to sunset OpenCensus in OpenTelemetry because OTel replaces OpenCensus", or is there a definite cost in keeping it around and maintaining it in this repo and the Collector builds?
> 
> We discussed this at the SIG meeting this morning. Please look at the notes/watch the recording for context.
> 
> The OpenCensus project predates OpenTelemetry and the maintainers made the commitment to maintain it when the collector project was established. After years of maintenance, there is no longer interest from maintainers and approvers to maintain this component, and we are deprecating it with the following possible outcomes:
> 
> * It gets removed from our source tree
> * Someone steps up to be a codeowner of the component
> 
> We have followed process ; after 6 months in deprecation state, this component is being removed.

I've read the notes and watched the video of the meeting. Thanks for discussing it. It seems clear to me that either someone steps forward to officially maintain the component, or it also gets removed from the repo at the end of the August 2025. In the meantime I will put some effort into producing my own Collector via OCB against a forked version of said OpenCensus Receiver.

**Comment by atoulme** [comment]
We are reaching the end of the month and no one has stepped up to volunteer and maintain those components. I will open a pull request to remove them.

**Comment by atoulme** [comment]
Last call. The PR to remove the code is open: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/42239

**Comment by alexchowle** [comment]
> > > It's removed from contrib, yes.
> > 
> > 
> > OK. Am I right in thinking that my only remaining option is to compile my own Collector against a frozen-in-time OpenCensus Receiver?
> 
> No, the receiver is still in the contrib tree and has been released as part of 0.132.0. You can build your own collector with that version.

- A related question - is it possible to build a custom collector using the **OpenTelemetry Collector Builder** whilst pinning the `opencensusreceiver` to v0.132.0? This caught my eye: https://github.com/open-telemetry/opentelemetry-collector/tree/main/cmd/builder#strict-versioning-checks

> The --skip-strict-versioning flag disables these versioning checks. This flag is available temporarily and will be removed in a future minor version.
